client/components/advanced-filters/index.js

```diff
@@ -0,0 +1,215 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import { __, sprintf } from '@wordpress/i18n';
+import { Component, Fragment } from '@wordpress/element';
+import { SelectControl, Button, FormTokenField, Dropdown } from '@wordpress/components';
+import { partial, findIndex, find, difference } from 'lodash';
+import PropTypes from 'prop-types';
+import Gridicon from 'gridicons';
+
+/**
+ * Internal dependencies
+ */
+import Card from 'components/card';
+import './style.scss';
+
+class AdvancedFilters extends Component {
+	constructor( props ) {
+		super( props );
+		this.state = {
+			match: props.config.matches[ 0 ],
+			activeFilters: [
+				/**
+				 * Example activeFilter
+				 * { key: ‘product’, rule: ‘includes’, value: [ ‘one’, ‘two’ ] }
+				 */
+			],
+		};
+
+		this.onMatchChange = this.onMatchChange.bind( this );
+		this.onFilterChange = this.onFilterChange.bind( this );
+		this.getSelector = this.getSelector.bind( this );
+		this.getAvailableFilterKeys = this.getAvailableFilterKeys.bind( this );
+		this.addFilter = this.addFilter.bind( this );
+		this.removeFilter = this.removeFilter.bind( this );
+		this.clearAll = this.clearAll.bind( this );
+	}
+
+	onMatchChange( value ) {
+		this.setState( {
+			match: find( this.props.config.matches, match => value === match.value ),
+		} );
+	}
+
+	onFilterChange( key, property, value ) {
+		const activeFilters = this.state.activeFilters.map( activeFilter => {
+			if ( key === activeFilter.key ) {
+				return Object.assign( {}, activeFilter, { [ property ]: value } );
+			}
+			return activeFilter;
+		} );
+
+		this.setState( { activeFilters } );
+	}
+
+	removeFilter( key ) {
+		const activeFilters = [ ...this.state.activeFilters ];
+		const index = findIndex( activeFilters, f => f.key === key );
+		activeFilters.splice( index, 1 );
+		this.setState( { activeFilters } );
+	}
+
+	getTitle() {
+		const { config } = this.props;
+		const { match } = this.state;
+		const label = sprintf( __( 'Orders Match %s Filters', 'wc-admin' ), match.label );
+		return (
+			<Fragment>
+				<span className="screen-reader-text">{ label }</span>
+				<span>{ __( 'Orders Match', 'wc-admin' ) }</span>
+				<SelectControl
+					className="woocommerce-advanced-filters__title-select"
+					options={ config.matches }
+					value={ match.value }
+					onChange={ this.onMatchChange }
+				/>
+				<span>{ __( 'Filters', 'wc-admin' ) }</span>
+			</Fragment>
+		);
+	}
+
+	getSelector( filter ) {
+		const filterConfig = this.props.config.filters[ filter.key ];
+		const { input } = filterConfig;
+		if ( ! input ) {
+			return null;
+		}
+		if ( 'SelectControl' === input.component ) {
+			return (
+				<SelectControl
```

Comment:
These selectors (this & FormTokenField) need labels - this has the same problem as the header does where we're trying to use casual English sentences. Actually -- how does this play out in other languages where verb order is different, RTL languages…?